File: src/main/java/net/cjsah/console/exceptions/Para2CommandException.java
package net.cjsah.console.exceptions;

import net.cjsah.console.command.StringReader;

public class Para2CommandException implements CommandExceptionType {
    private final Function function;

    public Para2CommandException(final Function function) {
        this.function = function;
    }

    public CommandException create(Object a, Object b) {
        return new CommandException(this, this.function.apply(a, b));
    }

    public CommandException createWithContext(StringReader reader, Object a, Object b) {
        return new CommandException(this, this.function.apply(a, b), reader.getString(), reader.getCursor());
    }

    interface Function {
        String apply(Object a, Object b);
    }
}

File: src/main/kotlin/net/cjsah/console/exceptions/Para2CommandException.kt
@file:Suppress("unused")

package net.cjsah.console.exceptions

import net.cjsah.console.command.StringReader

class Para2CommandException(private val function: Function) : CommandExceptionType {

    fun create(a: Any, b: Any) = CommandException(this, function.apply(a, b))

    fun createWithContext(reader: StringReader, a: Any, b: Any) = CommandException(this, function.apply(a, b), reader.getString(), reader.getCursor())

    fun interface Function {
        fun apply(a: Any, b: Any): String
    }
}